cmake/mission_build.cmake

```diff
@@ -0,0 +1,487 @@
+    "${osal_MISSION_DIR}/src/os/inc/*.h"
+    "${CMAKE_BINARY_DIR}/doc/osconfig-example.h")
+  string(REPLACE ";" " \\\n" MISSION_OSAL_HEADERFILES "${MISSION_OSAL_HEADERFILES}")
+
+  configure_file("${CFE_SOURCE_DIR}/cmake/cfe-usersguide.doxyfile.in"
+    "${CMAKE_BINARY_DIR}/doc/cfe-usersguide.doxyfile")
+
+  configure_file("${CFE_SOURCE_DIR}/cmake/osalguide.doxyfile.in"
+    "${CMAKE_BINARY_DIR}/doc/osalguide.doxyfile")
+
+  add_custom_target(mission-doc
+    doxygen mission-detaildesign.doxyfile
+    WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/doc")
+
+  add_custom_target(cfe-usersguide
+    doxygen cfe-usersguide.doxyfile
+    WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/doc")
+
+  add_custom_target(osalguide
+    doxygen osalguide.doxyfile
+    WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/doc")
+
+  # Pull in any application-specific mission-scope configuration
+  # This may include user configuration files such as cfe_mission_cfg.h,
+  # msgid definitions, or any other configuration/preparation that needs to
+  # happen at mission/global scope.
+  foreach(DEP_NAME ${MISSION_DEPS})
+    include("${${DEP_NAME}_MISSION_DIR}/mission_build.cmake" OPTIONAL)
+  endforeach(DEP_NAME ${MISSION_DEPS})
+
+  # Certain runtime variables need to be "exported" to the subordinate build, such as
+  # the specific arch settings and the location of all the apps.  This is done by creating
+  # a temporary file within the dir and then the subprocess will read that file and re-create
+  # variables out of them.  The alternative to this is to specify many "-D" parameters to the
+  # subordinate build but that would not scale well to many vars.
+  set(VARLIST
+    "MISSION_NAME"
+    "SIMULATION"
+    "MISSION_DEFS"
+    "MISSION_SOURCE_DIR"
+    "MISSION_BINARY_DIR"
+    "MISSIONCONFIG"
+    "MISSION_APPS"
+    "MISSION_PSPMODULES"
+    "MISSION_DEPS"
+    "ENABLE_UNIT_TESTS"
+  )
+  foreach(APP ${MISSION_DEPS})
+    list(APPEND VARLIST "${APP}_MISSION_DIR")
+  endforeach(APP ${MISSION_APPS})
+
+  foreach(SYSVAR ${TGTSYS_LIST})
+    list(APPEND VARLIST "BUILD_CONFIG_${SYSVAR}")
+  endforeach(SYSVAR ${TGTSYS_LIST})
+
+  set(MISSION_VARCACHE)
+  foreach(VARL ${VARLIST})
+    # It is important to avoid putting any blank lines in the output,
+    # This will cause the reader to misinterpret the data
+    if (NOT "${${VARL}}" STREQUAL "")
+      set(MISSION_VARCACHE "${MISSION_VARCACHE}${VARL}\n${${VARL}}\n")
+    endif (NOT "${${VARL}}" STREQUAL "")
+  endforeach(VARL ${VARLIST})
+  file(WRITE "${CMAKE_BINARY_DIR}/mission_vars.cache" "${MISSION_VARCACHE}")
+
+  generate_build_version_templates()
+
+  # Generate the tools for the native (host) arch
+  # Add all public include dirs for core components to include path for tools
+  include_directories(
+    ${core_api_MISSION_DIR}/fsw/inc
+    #${es_MISSION_DIR}/fsw/inc
+    #${evs_MISSION_DIR}/fsw/inc
+    #${fs_MISSION_DIR}/fsw/inc
+    #${sb_MISSION_DIR}/fsw/inc
+    #${tbl_MISSION_DIR}/fsw/inc
+    #${time_MISSION_DIR}/fsw/inc
```

Comment:
Should these commented out lines still be here?

Reply:
Also fixed in PR #1290